src/modules/itslearning/actions/base-action.ts

```diff
@@ -0,0 +1,64 @@
+import { XMLBuilder, XMLParser } from 'fast-xml-parser';
+
+import { DomainError, ItsLearningError } from '../../../shared/error/index.js';
+
+export type StatusInfo =
+    | {
+          codeMajor: 'failure';
+          severity: 'error';
+      }
+    | {
+          codeMajor: 'success';
+          severity: 'status';
+      };
+
+export type BaseResponse<BodyResponse> = {
+    Envelope: {
+        Header: {
+            syncResponseHeaderInfo: {
+                statusInfo: StatusInfo;
+            };
+        };
+
+        Body: BodyResponse;
+    };
+};
+
+export abstract class IMSESAction<ResponseBodyType, ResultType> {
+    protected readonly xmlBuilder: XMLBuilder = new XMLBuilder({ ignoreAttributes: false });
+
+    protected readonly xmlParser: XMLParser = new XMLParser({
```

Comment:
Think about injecting that maybe?

Reply:
The parser may need specific configuration per action (currently only the override for 'isArray', which tells the parser if a tag should always be treated as a list, even if only one entry is present)

Injecting an already constructed parser would also mean you can't easily construct instances of the different actions